server/webapp/WEB-INF/rails.new/app/controllers/admin/materials/pluggable_scm_controller.rb

```diff
@@ -0,0 +1,212 @@
+        assert_load :material, @subject
+      end
+    end
+
+    def new
+      scm = com.thoughtworks.go.domain.scm.SCM.new
+      scm.setPluginConfiguration(PluginConfiguration.new(params[:plugin_id], nil))
+      pluggable_scm = PluggableSCMMaterialConfig.new
+      pluggable_scm.setSCMConfig(scm)
+
+      assert_load :material, pluggable_scm
+      assert_load :meta_data_store, meta_data_store
+      render layout: false
+    end
+
+    def create
+      scm = com.thoughtworks.go.domain.scm.SCM.new
+      scm.setPluginConfiguration(PluginConfiguration.new(params[:plugin_id], '1'))
+      pluggable_scm = PluggableSCMMaterialConfig.new
+      pluggable_scm.setSCMConfig(scm)
+      assert_load :material, pluggable_scm
+
+      create_failure_handler = proc do |result, all_errors|
+        @errors = flatten_all_errors(all_errors)
+        @meta_data_store = meta_data_store
+        render :template => "/admin/materials/pluggable_scm/new", :status => result.httpCode(), :layout => false
+      end
+
+      save_popup(params[:config_md5], get_create_command, create_failure_handler, {:controller => '/admin/materials', :stage_parent => "pipelines", :current_tab => params[:current_tab]}) do
+        assert_load :pipeline, @node.pipelineConfigByName(CaseInsensitiveString.new(params[:pipeline_name]))
+        assert_load :material, @subject
+      end
+    end
+
+    def edit
+      assert_load :material, @pipeline.materialConfigs().getByFingerPrint(params[:finger_print])
+      assert_load :meta_data_store, meta_data_store
+      render layout: false
+    end
+
+    def update
+      material = @pipeline.materialConfigs().getByFingerPrint(params[:finger_print])
+      assert_load :material, material
+      assert_load :index, @pipeline.materialConfigs().indexOf(material)
+
+      create_failure_handler = proc do |result, all_errors|
+        @errors = flatten_all_errors(all_errors)
+        @meta_data_store = meta_data_store
+        render :template => "/admin/materials/pluggable_scm/edit", :status => result.httpCode(), :layout => false
+      end
+
+      save_popup(params[:config_md5], get_update_command, create_failure_handler, {:controller => '/admin/materials', :stage_parent => "pipelines", :current_tab => params[:current_tab]}) do
+        assert_load :pipeline, @node.pipelineConfigByName(CaseInsensitiveString.new(params[:pipeline_name]))
+        assert_load :material, @subject
+      end
+    end
+
+    def check_connection
+      scm = com.thoughtworks.go.domain.scm.SCM.new
+      scm.setPluginConfiguration(PluginConfiguration.new(params[:plugin_id], nil))
+      scm.setConfigAttributes(params[:material])
+
+      result = pluggable_scm_service.checkConnection(scm)
+      render json: { status: result.isSuccessful() ? 'success' : 'failure', messages: result.getMessages() }
+    end
+
+    private
+
+    def get_choose_existing_command
```

Comment:
Assignment Branch Condition size for get_choose_existing_command is too high. [27.07/15]